cmd/worker/internal/ratelimit/handler_test.go

```diff
@@ -0,0 +1,141 @@
+package ratelimit
+
+import (
+	"context"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbtest"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/internal/redispool"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Handle(t *testing.T) {
+	obsCtx := observation.TestContextTB(t)
+	logger := obsCtx.Logger
+	db := database.NewDB(logger, dbtest.NewDB(logger, t))
+	prefix := "__test__" + t.Name()
+	ctx := context.Background()
+	ten := int32(10)
+	url := "https://github.com/"
+	redisHost := "127.0.0.1:6379"
+	codeHost := &types.CodeHost{
+		Kind:                        extsvc.KindGitHub,
+		URL:                         url,
+		APIRateLimitQuota:           &ten,
+		APIRateLimitIntervalSeconds: &ten,
+		GitRateLimitQuota:           &ten,
+		GitRateLimitIntervalSeconds: &ten,
+	}
+	pool := &redis.Pool{
+		MaxIdle:     3,
+		IdleTimeout: 240 * time.Second,
+		Dial: func() (redis.Conn, error) {
+			return redis.Dial("tcp", redisHost)
+		},
+		TestOnBorrow: func(c redis.Conn, t time.Time) error {
+			_, err := c.Do("PING")
+			return err
+		},
+	}
+	assertValFromRedis := func(key string, expectedVal int32, kv redispool.KeyValue) {
```

Comment:
Nit, but I'd rewrite signature like this:

```suggestion
	assertValFromRedis := func(kv redispool.KeyValue, key string, expectedVal int32) {
```

to match the pattern of "receiver, args for receiver", which we have for other test helpers or functions in general.